pkg/controller/deployment_util.go

```diff
@@ -1220,3 +1291,45 @@ func statusUpdateRequired(old v1alpha1.MachineDeploymentStatus, new v1alpha1.Mac
 
 	return true
 }
+
+// MergeWithOverwriteAndFilter returns a new map containing keys and values from the `currentMap` map
+// that are not present in `oldMap`, combined with all keys and values from `newMap`.
+// If a key exists in both `currentMap` and `newMap`, the value from `newMap` is used.
+// This effectively merges the maps while overwriting and filtering as described.
+func MergeWithOverwriteAndFilter[T any](currentMap, oldMap, newMap map[string]T) map[string]T {
```

Comment:
- This is a very generically named method which has specific comments about oldIS.
- You have made this look like a utility function when this function makes assumptions. It assumes that old and new Map have same keys.
- Then you are half way there w.r.t using generics. You assume that the key is always a string instead of using `comparable` and value can be any Type. Where as the usage suggests that both key and values are always going to be strings.